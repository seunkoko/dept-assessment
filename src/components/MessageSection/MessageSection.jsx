// eslint-disable-next-line
import React from 'react';

import './MessageSection.scss';


const MessageSection = ({
  email,
  name,
  message,
  emailError,
  nameError,
  messageError,
  successMessage,
  handleOnChange,
  handleAlertClose,
  handleSubmitMessage
}) => {
	return (
    <>
      <div className="container-fluid message-section-style">
        {
          successMessage && <div className="alert alert-success message-alert-success" role="alert">
            {successMessage}
            <button
              type="button"
              className="close"
              aria-label="Close"
              onClick={handleAlertClose}
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
        }

        <div>
          <div className="question-style">
            <p>QUESTION?</p>
            <p>WE ARE HERE</p>
            <p>TO HELP!</p>
          </div>

          <form
            onSubmit={handleSubmitMessage}
          >
            <div className="form-style">
              <div>
                <div className="form-group input-container">
                  <label htmlFor="name">Name</label>
                  <input
                    className="form-control form-control-sm input-style"
                    id="name"
                    type="text"
                    name="name"
                    value={name}
                    onChange={handleOnChange}
                    aria-describedby="nameHelp"
                  />
                  <small id="nameHelp" className="form-text red-text">
                    {nameError}
                  </small>
                </div>

                <div className="form-group input-container">
                  <label htmlFor="email">Email</label>
                  <input
                    className="form-control form-control-sm email input-style"
                    id="email"
                    type="email"
                    name="email"
                    value={email}
                    onChange={handleOnChange}
                    aria-describedby="emailHelp"
                  />
                  <small id="emailHelp" className="form-text red-text">
                    {emailError}
                  </small>
                </div>

                <div className="form-group input-container">
                  <label htmlFor="message">Message</label>
                  <textarea
                    className="form-control form-control-sm input-style"
                    id="message"
                    type="text"
                    name="message"
                    value={message}
                    onChange={handleOnChange}
                    aria-describedby="messageHelp"
                  />
                  <small id="messageHelp" className="form-text red-text">
                    {messageError}
                  </small>
                </div>
              </div>

              <button
                type="submit"
                className="btn btn-send-message"
              >
                Send
              </button>
            </div>
          </form>
        </div>
      </div>
    </>
	);
};

export default MessageSection;
