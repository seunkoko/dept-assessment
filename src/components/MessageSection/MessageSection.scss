@import "../../stylesheets/variables";

.message-section-style {
  margin: 730px 0 250px 0;
  width: 80%;
  letter-spacing: 1px;
  padding: 60px 0 0 10%;
  
  > div:last-child {
    height: 250px;
    display: grid;
    grid-template-columns: 28% 70%;
    grid-column-gap: 2%;

    @media (max-width: $phone-width) {
      height: auto;
      display: flex;
      flex-direction: column;
    }
  }

  @media (max-width: $phone-width) {
    width: 100%;
    height: auto;
    margin: 570px 0 -30px 0;
    padding: 20px 0 0 0;
  }
}

#message {
  resize: none;
  height: 80px;
}

.question-style {
  padding: 30px 0;

  > p {
    margin: 0;
    width: fit-content;
    line-height: 2.2rem;
    font-family: 'Teko-Regular';
    font-size: $big-font;

    @media (max-width: $mobile-width) {
      font-size: 30px;
    }

    @media (max-width: $phone-width) {
      display: inline-block;
      margin: 30px 0 0 5px;
      font-size: 28px;
    }
  }

  @media (max-width: $phone-width) {
    padding: 0;
    margin: auto;
  }
}

.form-style {
  padding: 30px 0;

  > div {
    margin-bottom: 30px;
  }

  > div:first-child {
    display: grid;
    grid-template-columns: 48.5% 48.5%;
    grid-column-gap: 3%;

    @media (max-width: $phone-width) {
      margin-bottom: 0;
      display: flex;
      flex-direction: column;
    }
  }
}

.input-container {
  font-family: 'Teko-Regular';

  @media (max-width: $phone-width) {
    margin-bottom: 20px;
  }
}

.input-container:last-child {
  margin-top: 20px;

  @media (max-width: $phone-width) {
    margin-top: 0;
  }
}

.input-style {
  background: transparent;
  border-radius: 0;
  border: none;
  border-bottom: 1px solid $faint-gray;
  padding: 0;
  letter-spacing: 1.5px;
  font-family: 'Teko-Light';
}

.btn-send-message {
  @include button-style;

  width: 150px;
  background-color: $primary-blue;

  &:hover {
    background-color: $primary-white;
    color: $primary-blue;
    border: 1px solid $primary-blue;
  }
}

.message-alert-success {
  letter-spacing: 1px;
  font-family: 'Teko-Light';
}
