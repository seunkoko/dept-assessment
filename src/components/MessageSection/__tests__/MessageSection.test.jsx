import React from 'react';

import { shallow } from 'enzyme';

import MessageSection from '../MessageSection';

describe('MessageSection', () => {
	const component = shallow(<MessageSection />);

	it('should render MessageSection component', () => {
		expect(component).toMatchSnapshot();
	});
	
	// it('should add `Other values` when Enter is pressed', () => {
	// 	const tree = shallow(<component />);

	// });
});