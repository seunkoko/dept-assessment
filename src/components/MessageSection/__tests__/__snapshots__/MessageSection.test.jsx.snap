// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageSection should render MessageSection component 1`] = `
<Fragment>
  <div
    className="container-fluid message-section-style"
    id="contact"
  >
    <div>
      <div
        className="question-style"
      >
        <p>
          QUESTION?
        </p>
        <p>
          WE ARE HERE
        </p>
        <p>
          TO HELP!
        </p>
      </div>
      <form>
        <div
          className="form-style"
        >
          <div>
            <div
              className="form-group input-container"
            >
              <label
                htmlFor="name"
              >
                Name
              </label>
              <input
                aria-describedby="nameHelp"
                className="form-control form-control-sm input-style"
                id="name"
                name="name"
                type="text"
              />
              <small
                className="form-text red-text"
                id="nameHelp"
              />
            </div>
            <div
              className="form-group input-container"
            >
              <label
                htmlFor="email"
              >
                Email
              </label>
              <input
                aria-describedby="emailHelp"
                className="form-control form-control-sm email input-style"
                id="email"
                name="email"
                type="email"
              />
              <small
                className="form-text red-text"
                id="emailHelp"
              />
            </div>
            <div
              className="form-group input-container"
            >
              <label
                htmlFor="message"
              >
                Message
              </label>
              <textarea
                aria-describedby="messageHelp"
                className="form-control form-control-sm input-style"
                id="message"
                name="message"
                type="text"
              />
              <small
                className="form-text red-text"
                id="messageHelp"
              />
            </div>
          </div>
          <button
            className="btn btn-send-message"
            type="submit"
          >
            Send
          </button>
        </div>
      </form>
    </div>
  </div>
</Fragment>
`;
